### PR TITLE
Fix some minor startup issues for clarity and usefulness.

### PR DESCRIPTION
**Pull request by darobs**
It didn't really make sense that the default config.yaml had a different default home directory than the 
configuration in the packaging.  I also changed the default value for the connection string to make it clear that it needs to be updated.

Based on that default value, once configuration is loaded, if that default connection string is still set, we will generate explicit error message to indicate we need the user to configure IoT Edge.

### CODE REVIEW

File: edgelet/iotedged/src/error.rs [hunk 1]
```diff
@@ -54,6 +54,20 @@ pub enum ErrorKind {
     Parse,
     #[fail(display = "An http error occurred.")]
     Http,
+    #[cfg(target_os = "windows")]
+    #[fail(
+        display = "Edge device information is required.\n\
+        Please edit your config.yaml and provide information to connection the Edge to the IoT Hub.\n\
```
Comment (varunpuranik) [inline edgelet/iotedged/src/error.rs]: "to connect the Edge device to IoTHub. "
Or even better, "Please update the config.yaml and provide the IoTHub connection information."

File: edgelet/iotedged/src/error.rs [hunk 2]
```diff
@@ -54,6 +54,20 @@ pub enum ErrorKind {
...
+        See https://aka.ms/iot-edge-configure-windows"
+    )]
+    Unconfigured,
+    #[cfg(not(target_os = "windows"))]
+    #[fail(
+        display = "Edge device information is required.\n\
+        Please edit your config.yaml and provide information to connection the Edge to the IoT Hub.\n\
```
Comment (varunpuranik) [inline edgelet/iotedged/src/error.rs]: Same

File: edgelet/iotedged/src/error.rs [hunk 3]
```diff
@@ -54,6 +54,20 @@ pub enum ErrorKind {
     Parse,
     #[fail(display = "An http error occurred.")]
     Http,
+    #[cfg(target_os = "windows")]
+    #[fail(
+        display = "Edge device information is required.\n\
+        Please edit your config.yaml and provide information to connection the Edge to the IoT Hub.\n\
+        See https://aka.ms/iot-edge-configure-windows"
```
Comment (varunpuranik) [inline edgelet/iotedged/src/error.rs]: Nit: Suggest: "See aka.. for more details".

File: edgelet/iotedged/src/settings.rs [hunk 4]
```diff
@@ -167,14 +170,20 @@ where
...
         if let Some(file) = filename {
             config.merge(File::with_name(file).required(true))?;
         }
 
         config.merge(Environment::with_prefix("iotedge"))?;
 
-        let settings = config.try_into()?;
+        let settings: Self = config.try_into()?;
+
+        if let Provisioning::Manual(manual) = settings.provisioning() {
```
[the remark this answers is not in the document]
Comment (darobs) [reply edgelet/iotedged/src/settings.rs]: I was torn between the two places, I will move this logic.

File: edgelet/iotedged/src/error.rs [hunk 5]
```diff
@@ -57,15 +57,15 @@ pub enum ErrorKind {
     #[cfg(target_os = "windows")]
     #[fail(
         display = "Edge device information is required.\n\
-        Please edit your config.yaml and provide information to connection the Edge to the IoT Hub.\n\
-        See https://aka.ms/iot-edge-configure-windows"
+                   Please update the config.yaml and provide the IoTHub connection information.\n\
```
Comment (varunpuranik) [inline edgelet/iotedged/src/error.rs]: Does this formatting show up better? (Wondering about whitespace)

**Pull request merged**